### ISSUE TITLE
Whitelisted peers should never be banned

### ISSUE THREAD

**Issue by nomnombtc** [gandrewstone/bitcoin#581]
Hello,

I am running a node on TOR and already had problems in the past that BU would often ban my 127.0.0.1. But until recently it was enough to whitelist localhost (with whitelistforcerelay=0 so it does not relay all kinds of crap from TOR peers...)

But some of the recently introduced ban mechanism, like here for example: https://github.com/BitcoinUnlimited/BitcoinUnlimited/commit/4e55fdde78e3406aa02d535fd928e3fa5fac4f72#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR6633 do not check at all if a node is whitelisted, it just bans.

```
2017-05-14 05:03:33 Banning 127.0.0.1:49824 because initial headers were either not received or not received before the timeout
```

I fixed it for myself here https://github.com/nomnombtc/bitcoin/commit/ea4d25be540c0aab86ce1821d76abc4116dbdd8c , but not sure this is the correct way to do it. Imho running a node on TOR should also be possible without any whitelisting etc, maybe all those ban conditions also need to check if the offending address is configured as proxy (seems there is already something to do this with `IsProxy()`)

**Comment by ptschip** [comment]
you're right there are a couple of places where we use a reduced ban
time and have to explicitly invoke the ban or disconnect...your code fix
looks good for whitelisting.

can you submit a PR for it? if not i'll do it soon..thanks.

**Comment by nomnombtc** [comment]
@ptschip ok, done :)

**Comment by nomnombtc** [comment]
Not sure why this issue was not closed automatically, as #583 has been merged.
According to https://github.com/blog/1506-closing-issues-via-pull-requests I used one of the correct keywords. Anyway, closing the issue.

**Comment by ftrader** [comment]
TIL about this feature - but I've never seen it happen on the BU repo. Maybe there is a flag in the repo settings to allow it or not?
